L1Trigger/L1TGlobal/plugins/TriggerMenuParser.cc

```diff
@@ -974,14 +997,59 @@ void l1t::TriggerMenuParser::parseDeltaPhi_Cos_LUTS(std::map<std::string, tmeven
     lutName += "-";
     lutName += obj2;
     m_gtScales.setLUT_DeltaPhi(lutName,lut_delta_phi,prec1);
-        
+
     // Second Get the Cosh for this delta phi Set
     std::vector<long long> lut_cos;
     applyCos(val_delta_phi, n);
     setLut(lut_cos, val_delta_phi, prec2);
     m_gtScales.setLUT_Cos(lutName,lut_cos,prec2);
 
-}	
+}
+
+void l1t::TriggerMenuParser::parsePhi_Trig_LUTS(std::map<std::string, tmeventsetup::esScale> scaleMap, std::string obj, std::string func, unsigned int prec)
+{
+
+    using namespace tmeventsetup;
+
+    std::string scLabel = obj;
+    scLabel += "-PHI";
+
+    //This LUT does not exist in L1 Menu file, don't fill it
+    if( scaleMap.find(scLabel) == scaleMap.end() ) return;
+    if( func != "Sin" and func != "Cos") return;
+
+    const esScale* scale = &scaleMap.find(scLabel)->second;
+
+    const double step = scale->getStep();
+    const double range = scale->getMaximum() - scale->getMinimum();
+    const size_t n = std::ceil(range / step);
+    const size_t bitwidth = std::ceil(std::log10(n) / std::log10(2));
+
+    std::vector<double> array;
```

Comment:
these three lines can be more efficiently written as
std::vector<double> array(std::pow(2,bitwidth),0)